prompttools/experiment/experiments/stablediffusion_experiment.py

```diff
@@ -0,0 +1,163 @@
+except ImportError:
+    DiffusionPipeline = None
+try:
+    import torch
+except ImportError:
+    torch = None
+
+from prompttools.mock.mock import mock_stable_diffusion
+
+from .experiment import Experiment
+from .error import PromptExperimentException
+
+
+class StableDiffusionExperiment(Experiment):
+    r"""
+    Experiment for testing Stable Diffusion.
+
+    Args:
+    ----
+        hf_model_path (str): path to model on hugging face
+        use_auth_token (bool): boolean to determine if hf login is necessary [needed without GPU]
+        kwargs (dict): keyword arguments to call the model with
+    """
+
+    MODEL_PARAMETERS = ["hf_model_path", "prompt"]
+
+    CALL_PARAMETERS: List[str] = ["prompt"]
+
+    def __init__(
+        self,
+        hf_model_path: List[str],
+        prompt: List[str],
+        compare_images_folder: str,
+        use_auth_token: bool = False,
+        **kwargs: Dict[str, object],
+    ):
+        if DiffusionPipeline is None:
+            raise ModuleNotFoundError(
+                "Package `diffusers` is required to be installed to use this experiment."
+                "Please use `pip install diffusers` and \
+                `pip install invisible_watermark transformers accelerate safetensors` to install the package"
+            )
+        if torch is None:
+            raise ModuleNotFoundError(
+                "Package `torch` is required to be installed to use this experiment."
+                "Please use `pip install torch torchvision torchaudio` to install the package"
+            )
+        self.use_auth_token = use_auth_token
+        self.completion_fn = self.sd_completion_fn
+        self.compare_images_folder = compare_images_folder
+        if os.getenv("DEBUG", default=False):
+            self.completion_fn = mock_stable_diffusion
+        self.model_params = dict(hf_model_path=hf_model_path)
+
+        self.call_params = dict(prompt=prompt)
+        for k, v in kwargs.items():
+            self.CALL_PARAMETERS.append(k)
+            self.call_params[k] = v
+
+        self.all_args = self.model_params | self.call_params
+        super().__init__()
+
+    def prepare(self) -> None:
+        r"""
+        Combo builder.
+        """
+        self.model_argument_combos = [
+            dict(zip(self.model_params, val, strict=False)) for val in itertools.product(*self.model_params.values())
+        ]
+        self.call_argument_combos = [
+            dict(zip(self.call_params, val, strict=False)) for val in itertools.product(*self.call_params.values())
+        ]
+
+    def sd_completion_fn(
+        self,
+        **params: Dict[str, Any],
+    ):
+        r"""
+        Local model helper function to make request.
+        """
+        if cv2 is None:
+            raise ModuleNotFoundError(
+                "Package `cv2` is required to be installed to use this experiment."
+                "Please use `pip opencv-python` to install the package"
+            )
```

Comment:
nit: we probably want to raise this during `__init__` so the users can know ahead of time

Reply:
Good point- fixed. I left the guard in `experiment.py` under  `cv2_image_to_base64()` since the package is specific to image experiments